products/bridge/smart-contracts/contracts/core/Relayer.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.20;
+
+interface IRelayerEvents {
+    event Relayed(
+        uint indexed targetChainId,
+        address target,
+        bytes call,
+        uint gasLimit,
```

Comment:
What about the `gasPrice` - shouldn't that be included too?
For EIP1559 transactions, I think that would be the max base price and/or the tip/priority payment, but I actually can't find docs on how Eth mainnet RPCs accept them (https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendtransaction still only lists `gasPrice` which seems outdated for post-London? I may be looking in the wrong place). In any case at least one gas price parameter is likely necessary for all transaction types - unless you want the receiver side (validators for cross-shard, the Relayer contract for cross-chain) to set the fees themselves?

Reply:
I imagined it would probably get set by users. Should probably even be a parameter in the interface.

In the future shards might even deploy oracles to keep track of remote gas prices, but in the meantime having the user just supply it (basically functioning as a bootleg oracle for themselves) would be a simple solution - in the end the user pays the gas cost so it's their ultimate decision on what to try it with.

Reply:
Was wondering then, on contract execution, the contract can check that `tx.gasPrice` needs to match `gasPrice` set on the parameter? If the validator who submitted the transaction did set that `gasPrice` as such, it would pass signature verification?

Reply:
The `gasPrice` should be part of the event (and the signed data in the cross-chain case) and as you described above @WuBruno, the `Relayer` must check if `tx.gasprice` exceeds the limit and revert, penalizing the leader who submitted the transaction breaking the rules. However, EVM contracts can only access `tx.gasprice` which corresponds to `max_fee_per_gas` in case of EIP-1559 transactions. Therefore, in the cross-chain case we have no way to enforce `max_priority_fee_per_gas` specified by the caller contract. I would leave the tip out of the event in both the cross-chain and the cross-shard case.

Reply:
In the cross-chain case it's actually the surrogate contract paying the gas that is refunded to the validator submitting the transaction on the target chain. If the user who submitted the original transaction on the source case supplied the `gasPrice` on the target chain, they could drain the surrogate's deposit.

Reply:
We could use the Zilliqa validators as oracles instead, so that their supermajority signs updates to the `gasPrice` and the leader among them informs the relayer/gateway contract on the source chain.